spec.bs

```diff
@@ -1387,6 +1478,227 @@ To <dfn>build an interest group passed to generateBid</dfn> given an [=interest
   1. Return |igGenerateBid|.
 </div>
 
+<div algorithm>
+To <dfn>validate and convert additional bids</dfn> given an [=auction config=] |auctionConfig|, an
+[=auction config=]-or-null |topLevelAuctionConfig|, and a [=map=] |negativeTargetInfo| whose
+[=map/keys=] are [=tuples=] consisting of an [=origin=] and a [=string=], and whose [=map/values=]
+are [=tuples=] consisting of an [=origin=] and a [=byte sequence=]:
+
+  1. Let |auctionNonce| be |auctionConfig|'s [=auction config/auction nonce=].
+  1. Let |capturedAdditionalBidsHeaders| be [=this=]'s [=relevant global object=]'s
+    [=associated Document's=] [=node navigable's=] [=traversable navigable's=]
+    [=captured additional bids headers=].
+  1. Let |additionalBids| be a new [=list=] of [=additional bids=].
+  1. [=list/For each=] |encodedSignedAdditionalBid| of |capturedAdditionalBidsHeaders|[|auctionNonce|]:
+    1. Let |signedAdditionalBid| be the result of running [=forgiving-base64 decode=] with
+      |encodedSignedAdditionalBid|.
+    1. If |signedAdditionalBid| is failure, then [=iteration/continue=].
+    1. Let |additionalBid| be the result of running [=parse a signed additional bid=] given
+      |signedAdditionalBid|, |auctionConfig|, |topLevelAuctionConfig|, and |negativeTargetInfo|. 
+    1. If |additionalBid| is not null:
+      1. [=list/Append=] |additionalBid| to |additionalBids|.
+  1. Return |additionalBids|.
+</div>
+
+<div algorithm>
+To <dfn>parse a signed additional bid</dfn> given a [=byte sequence=] |signedAdditionalBid|, an
+[=auction config=] |auctionConfig|, an [=auction config=]-or-null |topLevelAuctionConfig|, and a
+[=map=] |negativeTargetInfo| whose [=map/keys=] are [=tuples=] consisting of an [=origin=] and a

```

Comment:
At this point, having seen this long map type written out at least three times in full, I'd recommend defining this as a separate "type" elsewhere in the spec, and referring to it everywhere it is currently duplicated. That would probably improve readability, and be quicker. Ideally this type would be defined in some kind of "Protected Audience & Negative Targeting" section, which I think would be good to include anyways for some extra context around what that is and how PA deals with it.